wagtail/wagtailadmin/static_src/wagtailadmin/js/page-editor.js

```diff
@@ -391,6 +391,28 @@ $(function() {
         insertRichTextDeleteControl(this);
     });
 
+    /* Make user confirm before leaving the editor if there are unsaved changes */
+    var initialData = $('#page-edit-form').serialize();
+    function dataHasChanged() {
+        return $('#page-edit-form').serialize() != initialData;
+    }
+
+    window.addEventListener('beforeunload', function(event) {
+        // Ignore if the user clicked one of the submit buttons
+        var trigger = $(event.target.activeElement);
+        var triggeredBySubmitButton = false;
+        $('#page-edit-form button[type="submit"]').each(function() {
+            if ($(this).is(trigger)) {
+                triggeredBySubmitButton = true;
+            }
+        });
+
+        if (!triggeredBySubmitButton && dataHasChanged()) {
+            event.returnValue = '\o/';
+            return '\o/';
```

Comment:
Chrome (47.0.2526.111) shows this text, so I think we need to write something human readable. Localizations also required in my opinion.

<img width="562" alt="screen shot 2016-01-28 at 13 31 11" src="https://cloud.githubusercontent.com/assets/509198/12641924/9cae969e-c5c3-11e5-8590-68d69cf34466.png">